### DISCUSSION TITLE
Splitting page-data.json into chunks for large site (10k products in category)

### DISCUSSION THREAD

**Discussion by creotip** [gatsbyjs/gatsby#31787] [Help]
We are developing an e-commerce website with gatsby (3.6.1).
Currently, we have some categories that include thousands of products. Those products are injected via graphql queries.
One category fetching 687kb page-data.json, which is bad for performance and user experience:

Here is an example. Go to clothing > shop all
https://60b5aabda5db9a00085b57fd--gatsbys-adika-torefront.netlify.app/

Is there solution for this ?... may be splitting page-data.json into chunks ?

**Comment by KyleAMathews** [comment]
Pagination is generally the best way to handle this.

**Comment by KyleAMathews** [comment]
Pagination is generally the best way to handle this.
